dev/com.ibm.ws.cdi.extension_fat/fat/src/com/ibm/ws/cdi/extension/apps/invocationContext/MyNonBindingInterceptor.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.cdi.extension.apps.invocationContext;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.lang.annotation.Annotation;
+import java.util.Set;
+
+import javax.annotation.Priority;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+
+import com.ibm.ws.cdi.internal.interfaces.CDIUtils;
+
+@Priority(1)
+@Interceptor
+@MyNonBindingInterceptorBinding
+public class MyNonBindingInterceptor {
+
+    @AroundInvoke
+    public Object aroundInvokeMethod(InvocationContext ctx) throws Exception {
+        //In a real app this wouldn't be possible, but for now we're just testing this utility method.
+        Set<Annotation> interceptorBindings = CDIUtils.getInterceptorBindingsFromInvocationContext(ctx);
+
+        boolean foundBindingAnnotation = false;
+        boolean foundNonBindingAnnotation = false;
+
+        assertEquals(2, interceptorBindings.size());
+        for (Annotation ann : interceptorBindings) {
+            if (MyInterceptorBinding.class == ann.annotationType()){
+                if (foundBindingAnnotation) {
+                    fail("found the binding annotation twice!");
+                }
+                foundBindingAnnotation = true;
+            }
+            if (MyNonBindingInterceptorBinding.class == ann.annotationType()){
+                if (foundNonBindingAnnotation) {
+                    fail("found the non binding annotation twice!");
+                }
+                foundNonBindingAnnotation = true;
+            }
+        }
+        assertTrue(foundNonBindingAnnotation);
+        assertTrue(foundBindingAnnotation);
```

Comment:
This isn't the behaviour I'd expected.

I expected `getInterceptorBindingsFromInvocationContext()` would return just `@MyNonBindingInterceptorBinding` here because that's the annotation used to bind this interceptor.

`@MyInterceptorBinding` isn't used to bind this interceptor so I didn't expect it to be returned.

Reply:
I'll double check, but this test passed.